aten/src/ATen/native/xnnpack/Clamp.cpp

```diff
@@ -0,0 +1,194 @@
+#ifdef USE_XNNPACK
+
+#include <ATen/native/xnnpack/Allocator.h>
+#include <ATen/native/xnnpack/Common.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+namespace clamp {
+
+struct Context final {
+  Operator clamp_op;
+};
+
+namespace {
+
+// Supports NHWC and NCHW FP32
+//  - relu
+//  - relu6
+//  - leaky relu
+//  - hard tanh
+//  - or any other operator that can be expressed as a clamp
+
+bool available(
+    const int64_t input_channels,
+    const int64_t output_channels,
+    const Scalar output_min,
+    const Scalar output_max) {
+         // XNNPACK
+  return xnnpack::internal::available() &&
+         // Input == Output
+         (input_channels == output_channels) &&
+         // Output Min / Max
+         (output_min.isIntegral(true) || output_min.isFloatingPoint()) &&
+         (output_max.isIntegral(true) || output_max.isFloatingPoint()) &&
+         (output_max.to<float>() > output_min.to<float>());
+}
+
+Context create(
+    const int64_t input_channels,
+    const int64_t output_channels,
+    const Scalar output_min,
+    const Scalar output_max) {
+  TORCH_CHECK(
+      available(input_channels, output_channels, output_min, output_max),
+      "xnnpack::clamp not available!");
+
+  xnn_operator_t clamp_op{};
+
+  const xnn_status create_status = xnn_create_clamp_nc_f32(
+      input_channels,         // channels,
+      input_channels,         // input_pixel_stride
+      output_channels,        // output_pixel_stride
+      output_min.to<float>(), // output_min
+      output_max.to<float>(), // output_max
+      0u,                     // flags
+      &clamp_op);             // operator
+
+  TORCH_INTERNAL_ASSERT(
```

Comment:
It should be TORCH_CHECK, not TORCH_INTERNAL_ASSERT - failure of the above op will be triggered by user passing invalid data. I think @ezyang left similar comment somewhere.

Reply:
Agreed.

Reply:
Removing Clamp.